lib/shopify_api/rest/request.ex

```diff
@@ -1,82 +1,76 @@
 
-  @transport "https://"
-  if Mix.env() == :test do
-    @transport "http://"
-  end
+  # Use HTTP in test for Bypass, HTTPS in all other environments
+  @transport if Mix.env() == :test, do: "http://", else: "https://"
 
   @http_receive_timeout Application.get_env(:shopify_api, :http_timeout)
 
-  @spec get(AuthToken.t(), String.t()) ::
-          {:ok, Response.t() | AsyncResponse.t()} | {:error, Error.t()}
-  def get(auth, path), do: shopify_request(:get, url(auth, path), "", headers(auth), auth)
-
-  @spec put(AuthToken.t(), String.t(), map()) ::
-          {:ok, Response.t() | AsyncResponse.t()} | {:error, Error.t()}
-  def put(auth, path, object),
-    do: shopify_request(:put, url(auth, path), Poison.encode!(object), headers(auth), auth)
-
-  @spec post(AuthToken.t(), String.t(), map()) ::
-          {:ok, Response.t() | AsyncResponse.t()} | {:error, Error.t()}
-  def post(auth, path, object \\ %{}),
-    do: shopify_request(:post, url(auth, path), Poison.encode!(object), headers(auth), auth)
-
-  @spec delete(AuthToken.t(), String.t()) ::
-          {:ok, Response.t() | AsyncResponse.t()} | {:error, Error.t()}
-  def delete(auth, path), do: shopify_request(:delete, url(auth, path), "", headers(auth), auth)
-
-  defp shopify_request(action, url, body, headers, token) do
-    Throttled.request(
-      fn ->
-        {time, response} =
-          :timer.tc(fn ->
-            request(action, url, body, headers, recv_timeout: @http_receive_timeout)
-          end)
-
-        log_request(action, url, time, response)
-
-        case response do
-          {:ok, %{status_code: status} = response} when status >= 200 and status < 300 ->
-            # TODO probably have to return the response here if we want to use the headers
-            ThrottleServer.update_api_call_limit(response, token)
-            {:ok, fetch_body(response)}
-
-          {:ok, response} ->
-            ThrottleServer.update_api_call_limit(response, token)
-            {:error, response}
-
-          response ->
-            {:error, response}
-        end
-      end,
-      token
+  ## Public Interface
+
+  def perform(%AuthToken{} = token, method, path, body \\ "") do
+    url = url(token, path)
+    headers = headers(token)
+
+    response =
+      Throttled.request(
+        fn -> request(method, url, body, headers) end,
+        token
+      )
+
+    case response do
+      {:ok, %{status_code: status} = response} when status >= 200 and status < 300 ->
+        # TODO probably have to return the response here if we want to use the headers
+        {:ok, fetch_body(response)}
+
+      {:ok, response} ->
+        {:error, response}
+
+      response ->
+        {:error, response}
+    end
+  end
+
+  def version do
+    Keyword.get(
+      Application.get_env(:shopify_api, ShopifyAPI.REST) || [],
+      :api_version,
+      @default_api_version
     )
   end
 
-  defp log_request(action, url, time, response) do
+  ## HTTPoison Overrides
+
```

Comment:
🍎 should we add `@impl` directives here now that they don't stomp on the HTTPoison funcs?